### PR TITLE
feat(ui): useDropdownable composable and close-on-value-update prop

### LINKED ISSUE #3459: Date Input `close-on-value-updated` prop
**Vuestic-ui version:** 1.6.5
**Steps to reproduce**
Set date input behavior using close-on-content-click.
**What is the expected behavior?**
You should be able to cycle through months using the arrows and clicking on a date should close the popup and select the date.
**What is the current behavior?**
https://codesandbox.io/p/sandbox/nice-darwin-iqjx2m
You cannot cycle through the months, but clicking on a date does close the popup and select the date.

### PR DESCRIPTION
**Pull request by m0ksem**
closes #3459

Moved all stuff for components like VaTimeInput, VaDateInput, VaSelect to separate composable. Might be used later in VaColorPicker, etc.

### CODE REVIEW

**Comment by m0ksem** [comment]
Check if it works with manual-input pickers.

File: packages/ui/src/composables/useDropdownable.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import { VaDropdown } from '../components/va-dropdown'
+
+const VaDropdownProps = extractComponentProps(VaDropdown,
+  ['innerAnchorSelector', 'stateful', 'keyboardNavigation', 'modelValue', 'ariaLabel'],
+)
+
+export const useDropdownableProps = {
+  ...VaDropdownProps,
+  modelValue: {},
+  closeOnValueUpdate: { type: Boolean, default: false },
```
Comment (asvae) [inline packages/ui/src/composables/useDropdownable.ts:13]: I think this should be 3 states: `false`, `true`, `null`. 
* `false` - would never close on select
* `true` - close on select
* `null` - default component behavior

Here's couple of examples why this matters.

For select:
* single select should close on input by default
* multiselect should **not** close on input by default

For time picker
* when you have only one column - we need to close on input by default
* when you have multiple columns - we shoudn't close by default

Probably also applicable to date picker - where for single non range input we want to close by default.

Comment (asvae) [reply packages/ui/src/composables/useDropdownable.ts:13]: Also I read the user issue and it seems their concern was with defaults, not with ability to force closing on select.

**Review by asvae — CHANGES_REQUESTED** [review]
Let's spend more time on this.

File: packages/ui/src/composables/useDropdownable.ts (hunk 1 above)
Comment (m0ksem) [reply packages/ui/src/composables/useDropdownable.ts:13]: Defaults can be configured trough GlobalConfig.

Comment (asvae) [reply packages/ui/src/composables/useDropdownable.ts:13]: @m0ksem they can't be if you need different defaults depending on other props.

File: packages/ui/src/components/va-date-input/VaDateInput.demo.vue [hunk 2]
```diff
@@ -192,6 +192,10 @@
       <va-button @click="focusDateInput">Focus</va-button>
       <va-button @click="blurDateInput">Blur</va-button>
     </VbCard>
+
+    <VbCard title="Close on value update">
+      <va-date-input v-model="value" close-on-value-update />
```
[the remark this answers is not in the document]
Comment (asvae) [reply packages/ui/src/components/va-date-input/VaDateInput.demo.vue:197]: Do we have a reason to prefer "close-on-update" to "close-on-select"?

[the remark this answers is not in the document]
Comment (m0ksem) [reply packages/ui/src/components/va-date-input/VaDateInput.demo.vue]: The only that it behaves on value update, not actual select. So if value changed outside of component (no select from user) dropdown is closed.

**Pull request merged**